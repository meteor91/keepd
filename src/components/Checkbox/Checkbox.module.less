.checkbox {
  display: inline-flex;
  font-size: 1rem;
  line-height: 1.7;
  cursor: pointer;
  &-icon {
    position: relative;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    padding: 0 5px;
    cursor: inherit;
  }

  &-input {
    z-index: 1;
    opacity: 0;
    position: absolute;
    top: 0;
    left: 0;
    cursor: inherit;
  }
  &-label {
    cursor: inherit;
  }
}