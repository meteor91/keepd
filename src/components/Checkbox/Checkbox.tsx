import React from 'react';
import style from './Checkbox.module.less';
import { CheckboxInactiveIcon } from '../../internal/svg-icons/CheckboxInactiveIcon';
import { CheckboxActiveIcon } from '../../internal/svg-icons/CheckboxActiveIcon';

export interface CheckboxProps {
    checked: boolean;
    name: string;
    label: string;
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Checkbox: React.FC<CheckboxProps> = (props) => {
    const { checked, name, label, onChange } = props;
    return (
        <label
            className={style.checkbox}
        >
            <span className={style['checkbox-icon']}>
                <input
                    name={name}
                    className={style['checkbox-input']}
                    onChange={onChange}
                    checked={checked}
                    type="checkbox"
                />
                {checked
                    ? <CheckboxActiveIcon/>
                    : <CheckboxInactiveIcon/>
                }
            </span>

            <span className={style['checkbox-label']}>{label}</span>
        </label>
    );
};
