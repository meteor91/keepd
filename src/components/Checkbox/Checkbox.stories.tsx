import React, {useState} from "react";
import type {Meta, StoryObj} from '@storybook/react';

import {Checkbox} from './Checkbox';

const meta: Meta<typeof Checkbox> = {
    /* ðŸ‘‡ The title prop is optional.
     * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
     * to learn how to generate automatic titles
     */
    title: 'Checkbox',
    component: Checkbox,
};

export default meta;
type Story = StoryObj<typeof Checkbox>;


export interface Props {
    label: string;
    name: string;
    value?: boolean;
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Preview = (props: Props) => {
    const {value: initialValue, ...rest} = props;

    const [value, setValue] = useState(initialValue ?? false);

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setValue(event.target.checked);
    }

    return (
        <Checkbox
            onChange={handleChange}
            checked={value}
            {...rest}
        />
    )
}

export const Default: Story = {
    render: () => <Preview label="Checkbox" name="checkbox" value={true}/>,
};
