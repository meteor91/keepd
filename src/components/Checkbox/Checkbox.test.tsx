import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Checkbox, type CheckboxProps } from './Checkbox';

describe('Checkbox', () => {
    const onChangeMock = jest.fn();

    const renderCheckbox = (props: CheckboxProps) => {
        return render(<Checkbox {...props} />);
    };

    it('should render correct', () => {
        const { container } = renderCheckbox({
            checked: true,
            name: 'checkbox',
            label: 'checkbox',
            onChange: onChangeMock,
        });

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should invoke onChange when change event fired', () => {
        renderCheckbox({
            checked: true,
            name: 'checkbox',
            label: 'checkbox',
            onChange: onChangeMock,
        });

        fireEvent.click(screen.getByLabelText('checkbox'));

        expect(onChangeMock).toHaveBeenCalledTimes(1);
    });
});
