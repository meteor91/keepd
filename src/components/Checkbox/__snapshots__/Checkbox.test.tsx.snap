// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correct 1`] = `
<label
  class="checkbox"
>
  <span
    class="checkbox-icon"
  >
    <input
      checked=""
      class="checkbox-input"
      name="checkbox"
      type="checkbox"
    />
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#1B4DFF"
        height="16"
        rx="4"
        width="16"
      />
      <path
        d="M11.4375 5.8125L7.0625 10.1875L4.875 8"
        stroke="white"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.53"
      />
    </svg>
  </span>
  <span
    class="checkbox-label"
  >
    checkbox
  </span>
</label>
`;
